### PR TITLE
Drop PG12 support

### PR DESCRIPTION
**Pull request by agedemenli**
DESCRIPTION: Drops postgres 12 support

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -18,7 +15,7 @@ parameters:
     default: '14.0'
   upgrade_pg_versions:
     type: string
-    default: '12.8-13.4-14.0'
+    default: '13.4-14.0'
```
Comment (hanefi) [inline .circleci/config.yml:18]: We require a docker image with a tag that is identical to this field. Did you create a new docker image for this to be valid?

File: .circleci/config.yml [hunk 2]
```diff
@@ -7,9 +7,6 @@ parameters:
   image_suffix:
     type: string
     default: '-v2021_10_27'
```
Comment (hanefi) [inline .circleci/config.yml:9]:
```suggestion
    default: '-dev-1e3a9bd'
```

We need to use the updated images here.

Currently we have a bug that is about to be fixed https://github.com/citusdata/the-process/pull/73

Once that lands, we can start using the suffixes without the `-dev` part

File: .circleci/config.yml (hunk 1 above)
Comment (hanefi) [reply .circleci/config.yml:18]: I created one in https://github.com/citusdata/the-process/pull/72

**Comment by onderkalaci** [comment]
Do we need to follow similar removals like here? https://github.com/citusdata/citus/pull/4799

Or, are we going to have a follow up PR for such changes?

**Comment by JelteF** [comment]
> Or, are we going to have a follow up PR for such changes?

Yeah the idea is to do that in a later PR, so that we can quickly merge this for the beta release. (afaik)

**Review by hanefi — APPROVED** [review]
Please squash, rebase and open a PR on enterprise as well

File: .circleci/config.yml [hunk 3]
```diff
@@ -783,33 +697,13 @@ workflows:
...
           old_pg_major: 13
           new_pg_major: 14
           image_tag: '<< pipeline.parameters.upgrade_pg_versions >>'
           requires: [build-13, build-14]
 
-      - test-citus-upgrade:
-          name: test-12_check-citus-upgrade
-          pg_major: 12
-          image_tag: '<< pipeline.parameters.pg12_version >>'
-          requires: [build-12]
```
[the remark this answers is not in the document]
Comment (hanefi) [reply .circleci/config.yml]: For the time being, we create citus-upgrade images for a single PG version. Maybe we should introduce multiple of these after making the necessary changes in the `the-process` repo.

**Pull request merged**